strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/io/ArtifactOutputStream.java

```diff
@@ -0,0 +1,31 @@
+package org.carlspring.strongbox.io;
+
+import java.io.FilterOutputStream;
+import java.io.OutputStream;
+
+import org.carlspring.strongbox.artifact.coordinates.ArtifactCoordinates;
+
+/**
+ * This OutputStream wraps a source stream from different Storage types (File System, AWS, JDBC, etc.).
+ * 
+ * @author Sergey Bespalov
+ *
+ */
+public class ArtifactOutputStream
+        extends FilterOutputStream
```

Comment:
This class should extend `MultipleDigestOutputStream` instead, as it should be able to produce checksums.